apps/design/backend/src/app.test.ts

```diff
@@ -894,3 +903,113 @@ test('CDF exports', async () => {
     ]),
   ]).toHaveLength(1);
 });
+
+test('getBallotPreviewPdf returns a ballot pdf for precinct with splits', async () => {
+  const baseElectionDefinition =
+    electionPrimaryPrecinctSplitsFixtures.readElectionDefinition();
+  const { apiClient } = await setupApp();
+
+  const electionId = (
+    await apiClient.loadElection({
+      electionData: baseElectionDefinition.electionData,
+    })
+  ).unsafeUnwrap();
+  const { ballotStyles, precincts } = await apiClient.getElection({
+    electionId,
+  });
+
+  const precinct = assertDefined(precincts.find((p) => hasSplits(p)));
+  const split = precinct.splits[0];
+  const ballotStyle = assertDefined(
+    ballotStyles.find((style) => {
+      const matchingSplit = style.precinctsOrSplits.find(
+        (p) => p.precinctId === precinct.id && p.splitId === split.id
+      );
+      return !!matchingSplit && style.languages.includes(LanguageCode.ENGLISH);
+    })
+  );
+
+  const result = (
+    await apiClient.getBallotPreviewPdf({
+      electionId,
+      precinctId: precinct.id,
+      ballotStyleId: ballotStyle.id,
+      ballotType: BallotType.Precinct,
+      ballotMode: 'test',
+      splitId: split.id,
+    })
+  ).unsafeUnwrap();
+
+  await expect(result).toMatchPdfSnapshot();
+});
+
+test('getBallotPreviewPdf returns a ballot pdf for precinct with no split', async () => {
+  const baseElectionDefinition =
+    electionFamousNames2021Fixtures.readElectionDefinition();
+  const { apiClient } = await setupApp();
+
+  const electionId = (
+    await apiClient.loadElection({
+      electionData: baseElectionDefinition.electionData,
+    })
+  ).unsafeUnwrap();
+  const { ballotStyles, precincts } = await apiClient.getElection({
+    electionId,
+  });
+
+  function hasDistrictIds(
+    precinct: Precinct
+  ): precinct is PrecinctWithoutSplits {
+    return 'districtIds' in precinct && precinct.districtIds.length > 0;
+  }
+
+  const precinct = assertDefined(precincts.find((p) => hasDistrictIds(p)));
+
+  const result = (
+    await apiClient.getBallotPreviewPdf({
+      electionId,
+      precinctId: precinct.id,
+      ballotStyleId: assertDefined(
+        ballotStyles.find(
+          (style) =>
+            style.districtIds.includes(precinct.districtIds[0]) &&
+            style.languages.includes(LanguageCode.ENGLISH)
+        )
+      ).id,
+      ballotType: BallotType.Precinct,
+      ballotMode: 'test',
+    })
+  ).unsafeUnwrap();
+
+  await expect(result).toMatchPdfSnapshot({ failureThreshold: 0.01 });
+});
```

Comment:
These snapshot tests actually render and add a few seconds to test runtime, increasing runtime from ~45 to ~50 seconds on my machine. Given the not-huge increase in absolute runtime I'm inclined to keep these in and convert to mocking the render after conversion to vitest.

Reply:
```
  ✓ getBallotPreviewPdf returns a ballot pdf for precinct with splits (1410 ms)
  ✓ getBallotPreviewPdf returns a ballot pdf for NH election with split precincts and additional config options (1348 ms)
  ✓ getBallotPreviewPdf returns a ballot pdf for precinct with no split (1430 ms)
  ...
  Time:        49.36 s
```